psydac/fem/grid.py

```diff
@@ -108,11 +115,16 @@ def __init__( self, space, start, end, *, quad_order=None, nderiv=1 ):
                     weights.append( glob_weights[k] )
                     indices.append( k )
                     ne += 1
-        
+
+
+        m = multiplicity if multiplicity>1 else 0
```

Comment:
this we need when the trial and the test spaces are diffrent and the parent multiplicity is `>1`, where we need more basis function than the usual

Reply:
What about the case where the current space is not a reduction, with `trial_space==test_space` and `pad==degree`? You get then

* if `multiplicity == 1`, then `m = 0`
* if `multiplicity > 1`, then `m = multiplicity`

So why is that? You should really explain the role of `m` in the code below, as I wrote before.

Reply:
it is used in the case of `pad==degree` because we can't know if the `trial_space==test_space`, here we compute the basis function of each space independently of each other.

Reply:
in the case where `multiplicity`>1 and the `test_space != trial_space`, we will need more basis functions (m more basis functions) otherwise the computation will be false, otherwise, the computation is the same as before